src/components/Modal/Modal.vue

```diff
@@ -40,15 +40,21 @@
     </template>
 
     <template #header>
-      <div v-if="icon" :class="`modal-icon-container modal-${iconColor}`">
-        <Icon :icon="icon" :size-override="35" />
+      <div
+        v-if="icon || slots.icon"
+        :class="`modal-icon-container modal-${iconColor}`"
+      >
+        <Icon v-if="icon" :icon="icon" :size-override="35" />
+        <slot name="icon" :size-override="35" />
       </div>
       <div>
         <p class="model-header-title">
           {{ header }}
+          <slot name="header" />
         </p>
```

Comment:
maybe something like this?
```suggestion
        <slot name="header">
          <p v-if="header" class="model-header-title">
            {{ header }}
          </p>
        </slot>
```

Reply:
Yeah this is almost exactly what I was thinking. I was torn though because I could see it get confusing why ones showing and one's not and figured the dev using the component could have more control in determining which are showing